core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -230,18 +222,92 @@
 import static io.trino.sql.tree.WindowFrame.Type.ROWS;
 import static java.lang.String.format;
 import static java.util.Collections.emptyList;
-import static java.util.Objects.requireNonNull;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
-import static org.testng.Assert.fail;
 
 public class TestSqlParser
 {
     private static final SqlParser SQL_PARSER = new SqlParser();
 
+    public static void assertGenericLiteral(String type)
+    {
+        assertThat(expression(type + " 'abc'")).isEqualTo(new GenericLiteral(location(1, 1), type, "abc"));
+    }
+
+    public static QuerySpecification createSelect123(NodeLocation queryLocation, NodeLocation columnLocation)
+    {
+        return simpleQuerySpecification(
+                queryLocation,
+                selectList(queryLocation, new LongLiteral(columnLocation, "123")));
+    }
+
+    private static ExistsPredicate exists(Query query)
+    {
+        return new ExistsPredicate(new SubqueryExpression(query));
+    }
+
+    private static ExistsPredicate exists(NodeLocation location, Query query)
```

Comment:
This should't have moved in this commit. Ideally, it should move to QueryUtil, but in a separate commit.